samples/01_basic_ping_bot/Program.cs

```diff
@@ -14,23 +14,31 @@ namespace _01_basic_ping_bot
     // - Here, under the 02_commands_framework sample
     // - https://github.com/foxbot/DiscordBotBase - a bare-bones bot template
     // - https://github.com/foxbot/patek - a more feature-filled bot, utilizing more aspects of the library
-    class Program
+    class Program : IDisposable
     {
-        private DiscordSocketClient _client;
+        private readonly DiscordSocketClient _client;
 
         // Discord.Net heavily utilizes TAP for async, so we create
         // an asynchronous context from the beginning.
         static void Main(string[] args)
-            => new Program().MainAsync().GetAwaiter().GetResult();
+        {
+            using (var program = new Program())
+                program.MainAsync().GetAwaiter().GetResult();
+        }
 
-        public async Task MainAsync()
+        public Program()
```

Comment:
As far as I'm aware, lots of initializer work has been done in constructors, so this is fine. And if the constructor throws (which I doubt it will), we're likely in a state where we cannot logically continue anyway, so for a sample, this is fine.